bonfire/config.py

```diff
@@ -16,7 +16,7 @@
 DEFAULT_ENV_PATH = get_config_path().joinpath("env")
 DEFAULT_SECRETS_DIR = get_config_path().joinpath("secrets")
 
-NAMESPACE_POOLS = ["default", "minimal", "managed-kafka", "real-managed-kafka"]
+with os.popen("oc get namespacepool --no-headers| cut -d' ' -f1") as f: NAMESPACE_POOL = f.readlines()
```

Comment:
A couple things to note here:
1. You should be able to call `from ocviapy import get_json` and then call `get_json("namespacepool")` function instead, which is essentially going to run `oc get namespacepool -o json` and return the result to you in a dictionary which you can then analyze.
2. I would not load the namespace pool on the top-level line of config here because that will cause this code to run any time bonfire runs. For example, even if you ran `bonfire --help` this could would get invoked.

Here is what I would suggest:
1. Get rid of `NAMESPACE_POOLS` from config and instead have just a `DEFAULT_NAMESPACE_POOL = "default"`

2. Add a `get_ns_pools` function to https://github.com/RedHatInsights/bonfire/blob/master/bonfire/openshift.py -- you can cache the result of this function (since the namespace pools are unlikely to change in the middle of a bonfire run) so use this same decorator: https://github.com/RedHatInsights/bonfire/blob/master/bonfire/openshift.py#L23 -- if there are no namespace pools found this function should probably return `[]`

3. I would then call the new function you wrote at this point: https://github.com/RedHatInsights/bonfire/blob/69b3f34b6f303fa182e7765fd329aa18d47785a1/bonfire/bonfire.py#L1002

to ensure that 'pool' is in the valid list of pools found on the cluster

4. The namespace pool list command will also need to be changed to call your new function instead of just printing `config.NAMESPACE_POOLS`:

https://github.com/RedHatInsights/bonfire/blob/69b3f34b6f303fa182e7765fd329aa18d47785a1/bonfire/bonfire.py#L868

5. And you will need to edit the 'type' here back to simply `type=str`

6. You'll have to update this unit test: https://github.com/RedHatInsights/bonfire/blob/69b3f34b6f303fa182e7765fd329aa18d47785a1/tests/test_bonfire.py#L271

Mock out the return value of your new 'get_ns_pools' function similar to how 'has_ns_operator' has been mocked out here: https://github.com/RedHatInsights/bonfire/blob/69b3f34b6f303fa182e7765fd329aa18d47785a1/tests/test_bonfire.py#L255